apps/internal/oauth/ops/authority/authority.go

```diff
@@ -103,36 +108,60 @@ func (r *TenantDiscoveryResponse) Validate() error {
 // ValidateIssuerMatchesAuthority validates that the issuer in the TenantDiscoveryResponse matches the authority.
 // This is used to identity security or configuration issues in authorities and the OIDC endpoint
 func (r *TenantDiscoveryResponse) ValidateIssuerMatchesAuthority(authorityURI string, aliases map[string]bool) error {
-
 	if authorityURI == "" {
 		return errors.New("TenantDiscoveryResponse: empty authorityURI provided for validation")
 	}
+	if r.Issuer == "" {
+		return errors.New("TenantDiscoveryResponse: empty issuer in response")
+	}
 
-	// Parse the issuer URL
 	issuerURL, err := url.Parse(r.Issuer)
 	if err != nil {
 		return fmt.Errorf("TenantDiscoveryResponse: failed to parse issuer URL: %w", err)
 	}
+	authorityURL, err := url.Parse(authorityURI)
+	if err != nil {
+		return fmt.Errorf("TenantDiscoveryResponse: failed to parse authority URL: %w", err)
+	}
+
+	// Fast path: exact scheme + host match
+	if issuerURL.Scheme == authorityURL.Scheme && issuerURL.Host == authorityURL.Host {
+		return nil
+	}
+
+	// Normalize for suffix checks (avoid reallocating repeatedly)
+	normalizedAuthority := strings.TrimSuffix(authorityURI, "/")
+	normalizedIssuer := strings.TrimSuffix(r.Issuer, "/")
 
-	// Even if it doesn't match the authority, issuers from known and trusted hosts are valid
+	// Allow authority to have extra path segments (issuer prefix)
+	if strings.HasSuffix(normalizedAuthority, normalizedIssuer) {
+		return nil
+	}
+
+	// Alias-based acceptance
 	if aliases != nil && aliases[issuerURL.Host] {
 		return nil
 	}
 
-	// Parse the authority URL for comparison
-	authorityURL, err := url.Parse(authorityURI)
-	if err != nil {
-		return fmt.Errorf("TenantDiscoveryResponse: failed to parse authority URL: %w", err)
+	// Direct trusted host
+	if TrustedHost(issuerURL.Host) {
+		return nil
 	}
 
-	// Check if the scheme and host match (paths can be ignored when validating the issuer)
-	if issuerURL.Scheme == authorityURL.Scheme && issuerURL.Host == authorityURL.Host {
+	// Regional authority pattern: <region>.<trustedHost>
+	// Accept if authority host is exactly trusted or ends with ".trustedHost" AND issuer host is trusted.
+	ah := authorityURL.Host
+	if TrustedHost(ah) && TrustedHost(issuerURL.Host) {
```

Comment:
this branch will never be true because L147 returns earlier for `TrustedHost(issuerURL.Host)`

Reply:
Nice catch, i updated it